modules/videoio/src/cap_mfx_common.hpp

```diff
@@ -176,10 +177,38 @@ inline void cleanup(T * &ptr)
 
 //==================================================================================================
 
+class MFXVideoSession_WRAP : public MFXVideoSession
```

Comment:
I believe, we should not embed factory behavior into objects (through inheritance). Separate classes for that should be more suitable.

Reply:
This class extends existing MediaSDK/oneVPL thin C++ wrapper which stores a session and provides analogs of C-functions via methods (https://github.com/Intel-Media-SDK/MediaSDK/blob/master/api/include/mfxvideo%2B%2B.h). `MFXVideoSession` already had this behavior (`Init`/`Close`), I've only added a new method to replace `Init`.

I'll think about moving loader out of this class and creating a new one (or singleton).

Reply:
Loader creation have been extracted.